### PR TITLE
Publish Scaladoc to Github pages.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This will aggregate all Scala documentations into one under the `api` folder on the
`gh-pages` branch. It will also copy all content from `./docs` to the `gh-pages`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,8 @@ before_cache:
   - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
   - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
 after_script:
-  - 'if [[ ! -z "$GIT_PASSWORD" ]]; then ./gradlew githubReport; fi'
+  - if [[ ! -z "$GIT_PASSWORD" ]]; then ./gradlew githubReport; fi
```
Comment (zen-dog) [inline .travis.yml]: Do you think, we should move this bash `if [[ ! -z ??? ]]` logic into the tasks?

File: .travis.yml [hunk 2]
```diff
@@ -5,7 +5,8 @@ before_cache:
...
+  - if [[ "$TRAVIS_BRANCH" = "master" -a "$TRAVIS_PULL_REQUEST" = "false" ]]; then ./gradlew gitPublishPush; fi
```
Comment (zen-dog) [inline .travis.yml]: Same idea here - how about passing the branch to the task: `gitPublishPush "$TRAVIS_BRANCH"`?

**Review by zen-dog — CHANGES_REQUESTED** [review]
See my comments

File: .travis.yml (hunk 1 above)
Comment (jeschkies) [reply .travis.yml]: I was wondering the same. Should I add `./gradlew ci` or `./gradlew after_script`? On the other side this is fairly Travis specific.

File: .travis.yml (hunk 2 above)
Comment (jeschkies) [reply .travis.yml]: 🤔 that's a good idea. The issues is that `TRAVIS_BRANCH` is the target branch for PRs unless the build is not a PR. I want this to be executed only when we are building master. Should we add the logic to Gradle as we have for Marathon?

```groovy
task master {
  description "Build target for master branch."
}
task pr {
  description "Build target for pull requests."
}
task ci {
  description "Build target for CIs such as Travis CI or Jenkins."
  if (isMasterBuild) dependsOn(master)
  else dependsOn(pr)
}
```

Comment (zen-dog) [reply .travis.yml]: Sgtm

Comment (alenkacz) [reply .travis.yml]: +1

**Review by alenkacz — APPROVED** [review]
Oh gradle... I like you so much.

Thanks Karsten, looking good (just the comment I +1d would be nice to change)!

**Review by meln1k — APPROVED** [review]

**Review by zen-dog — APPROVED** [review]
Nice!

**Comment by kensipe** [comment]
I reorganized the build file.   I would love to see us keep `build.gradle` clean if we can.   If you don't like... please revert.

**Review by kensipe — APPROVED** [review]

**Comment by kensipe** [comment]
if you revert... there are several redundant code snip its and a misspelling that we should resolve after the revert.

**Pull request merged**